From 3b5085171f2c9ec55de56b7ff793d90a9a511b8c Mon Sep 17 00:00:00 2001
From: Shivani Raval <sraval@habana.ai>
Date: Fri, 13 Jun 2025 13:04:44 +0300
Subject: [PATCH] [JSW-0] build: added common file for gaudi2 & 3

Added common script for gaudi3 and gaudi2 in common place
called gaudi.

Change-Id: I2a70222e39e80cfefe93da4e8faf98f7c60cb819
Signed-off-by: Shivani Raval <sraval@habana.ai>
---

diff --git a/board/habanalabs/gaudi/post-image-spi-arc_global.sh b/board/habanalabs/gaudi/post-image-spi-arc_global.sh
new file mode 100755
index 0000000..8ec13dc
--- /dev/null
+++ b/board/habanalabs/gaudi/post-image-spi-arc_global.sh
@@ -0,0 +1,15 @@
+#!/usr/bin/env bash
+set -e          # stop on any error in this script
+set -u          # stop on any reference to undefine variable
+set -o pipefail # stop on any error in pipeline
+
+source ${BR2_EXTERNAL}/board/habanalabs/common/post-image-spi-arc_global.sh
+
+pushd ${STAGE_DIR}
+
+	if [ -f "${KEY_BASE}/spi-flash-layout-validator.sh" ]; then
+		source ${KEY_BASE}/spi-flash-layout-validator.sh
+		validate_spi_flash_layout
+	fi
+
+popd
diff --git a/board/habanalabs/gaudi2_arc/gaudi2-post-image-spi-arc_global.sh b/board/habanalabs/gaudi2_arc/gaudi2-post-image-spi-arc_global.sh
deleted file mode 100755
index 8ec13dc..0000000
--- a/board/habanalabs/gaudi2_arc/gaudi2-post-image-spi-arc_global.sh
+++ /dev/null
@@ -1,15 +0,0 @@
-#!/usr/bin/env bash
-set -e          # stop on any error in this script
-set -u          # stop on any reference to undefine variable
-set -o pipefail # stop on any error in pipeline
-
-source ${BR2_EXTERNAL}/board/habanalabs/common/post-image-spi-arc_global.sh
-
-pushd ${STAGE_DIR}
-
-	if [ -f "${KEY_BASE}/spi-flash-layout-validator.sh" ]; then
-		source ${KEY_BASE}/spi-flash-layout-validator.sh
-		validate_spi_flash_layout
-	fi
-
-popd
diff --git a/board/habanalabs/gaudi3_arc/gaudi3-post-image-spi-arc_global.sh b/board/habanalabs/gaudi3_arc/gaudi3-post-image-spi-arc_global.sh
deleted file mode 100755
index 8ec13dc..0000000
--- a/board/habanalabs/gaudi3_arc/gaudi3-post-image-spi-arc_global.sh
+++ /dev/null
@@ -1,15 +0,0 @@
-#!/usr/bin/env bash
-set -e          # stop on any error in this script
-set -u          # stop on any reference to undefine variable
-set -o pipefail # stop on any error in pipeline
-
-source ${BR2_EXTERNAL}/board/habanalabs/common/post-image-spi-arc_global.sh
-
-pushd ${STAGE_DIR}
-
-	if [ -f "${KEY_BASE}/spi-flash-layout-validator.sh" ]; then
-		source ${KEY_BASE}/spi-flash-layout-validator.sh
-		validate_spi_flash_layout
-	fi
-
-popd
diff --git a/configs/gaudi2_arc_asic_defconfig b/configs/gaudi2_arc_asic_defconfig
index cf3cb21..545e626 100644
--- a/configs/gaudi2_arc_asic_defconfig
+++ b/configs/gaudi2_arc_asic_defconfig
@@ -11,7 +11,7 @@
 BR2_INIT_NONE=y
 BR2_SYSTEM_BIN_SH_NONE=y
 # BR2_TARGET_GENERIC_GETTY is not set
-BR2_ROOTFS_POST_IMAGE_SCRIPT="$(BR2_EXTERNAL_HABANALABS_PATH)/board/habanalabs/$(HL_DEV)/post-image-spi.sh $(BR2_EXTERNAL_HABANALABS_PATH)/board/habanalabs/$(HL_DEV)/gaudi2-post-image-spi-arc_global.sh"
+BR2_ROOTFS_POST_IMAGE_SCRIPT="$(BR2_EXTERNAL_HABANALABS_PATH)/board/habanalabs/$(HL_DEV)/post-image-spi.sh $(BR2_EXTERNAL_HABANALABS_PATH)/board/habanalabs/gaudi/post-image-spi-arc_global.sh"
 # BR2_PACKAGE_BUSYBOX is not set
 # BR2_PACKAGE_IFUPDOWN_SCRIPTS is not set
 # BR2_TARGET_ROOTFS_TAR is not set
diff --git a/configs/gaudi2_arc_defconfig b/configs/gaudi2_arc_defconfig
index cf3cb21..545e626 100644
--- a/configs/gaudi2_arc_defconfig
+++ b/configs/gaudi2_arc_defconfig
@@ -11,7 +11,7 @@
 BR2_INIT_NONE=y
 BR2_SYSTEM_BIN_SH_NONE=y
 # BR2_TARGET_GENERIC_GETTY is not set
-BR2_ROOTFS_POST_IMAGE_SCRIPT="$(BR2_EXTERNAL_HABANALABS_PATH)/board/habanalabs/$(HL_DEV)/post-image-spi.sh $(BR2_EXTERNAL_HABANALABS_PATH)/board/habanalabs/$(HL_DEV)/gaudi2-post-image-spi-arc_global.sh"
+BR2_ROOTFS_POST_IMAGE_SCRIPT="$(BR2_EXTERNAL_HABANALABS_PATH)/board/habanalabs/$(HL_DEV)/post-image-spi.sh $(BR2_EXTERNAL_HABANALABS_PATH)/board/habanalabs/gaudi/post-image-spi-arc_global.sh"
 # BR2_PACKAGE_BUSYBOX is not set
 # BR2_PACKAGE_IFUPDOWN_SCRIPTS is not set
 # BR2_TARGET_ROOTFS_TAR is not set
diff --git a/configs/gaudi2_arc_onix_defconfig b/configs/gaudi2_arc_onix_defconfig
index cf3cb21..545e626 100644
--- a/configs/gaudi2_arc_onix_defconfig
+++ b/configs/gaudi2_arc_onix_defconfig
@@ -11,7 +11,7 @@
 BR2_INIT_NONE=y
 BR2_SYSTEM_BIN_SH_NONE=y
 # BR2_TARGET_GENERIC_GETTY is not set
-BR2_ROOTFS_POST_IMAGE_SCRIPT="$(BR2_EXTERNAL_HABANALABS_PATH)/board/habanalabs/$(HL_DEV)/post-image-spi.sh $(BR2_EXTERNAL_HABANALABS_PATH)/board/habanalabs/$(HL_DEV)/gaudi2-post-image-spi-arc_global.sh"
+BR2_ROOTFS_POST_IMAGE_SCRIPT="$(BR2_EXTERNAL_HABANALABS_PATH)/board/habanalabs/$(HL_DEV)/post-image-spi.sh $(BR2_EXTERNAL_HABANALABS_PATH)/board/habanalabs/gaudi/post-image-spi-arc_global.sh"
 # BR2_PACKAGE_BUSYBOX is not set
 # BR2_PACKAGE_IFUPDOWN_SCRIPTS is not set
 # BR2_TARGET_ROOTFS_TAR is not set
diff --git a/configs/gaudi2_arc_palad_defconfig b/configs/gaudi2_arc_palad_defconfig
index cf3cb21..545e626 100644
--- a/configs/gaudi2_arc_palad_defconfig
+++ b/configs/gaudi2_arc_palad_defconfig
@@ -11,7 +11,7 @@
 BR2_INIT_NONE=y
 BR2_SYSTEM_BIN_SH_NONE=y
 # BR2_TARGET_GENERIC_GETTY is not set
-BR2_ROOTFS_POST_IMAGE_SCRIPT="$(BR2_EXTERNAL_HABANALABS_PATH)/board/habanalabs/$(HL_DEV)/post-image-spi.sh $(BR2_EXTERNAL_HABANALABS_PATH)/board/habanalabs/$(HL_DEV)/gaudi2-post-image-spi-arc_global.sh"
+BR2_ROOTFS_POST_IMAGE_SCRIPT="$(BR2_EXTERNAL_HABANALABS_PATH)/board/habanalabs/$(HL_DEV)/post-image-spi.sh $(BR2_EXTERNAL_HABANALABS_PATH)/board/habanalabs/gaudi/post-image-spi-arc_global.sh"
 # BR2_PACKAGE_BUSYBOX is not set
 # BR2_PACKAGE_IFUPDOWN_SCRIPTS is not set
 # BR2_TARGET_ROOTFS_TAR is not set
diff --git a/configs/gaudi3_arc_asic_defconfig b/configs/gaudi3_arc_asic_defconfig
index 75f86cd..bf8eb30 100644
--- a/configs/gaudi3_arc_asic_defconfig
+++ b/configs/gaudi3_arc_asic_defconfig
@@ -12,7 +12,7 @@
 BR2_INIT_NONE=y
 BR2_SYSTEM_BIN_SH_NONE=y
 # BR2_TARGET_GENERIC_GETTY is not set
-BR2_ROOTFS_POST_IMAGE_SCRIPT="$(BR2_EXTERNAL_HABANALABS_PATH)/board/habanalabs/common/post-image-fit-arc.sh $(BR2_EXTERNAL_HABANALABS_PATH)/board/habanalabs/$(HL_DEV)/gaudi3-post-image-spi-arc_global.sh $(BR2_EXTERNAL_HABANALABS_PATH)/board/habanalabs/common/post-image-agents.sh $(BR2_EXTERNAL_HABANALABS_PATH)/board/habanalabs/$(HL_DEV)/create-fake-fuse-image.sh"
+BR2_ROOTFS_POST_IMAGE_SCRIPT="$(BR2_EXTERNAL_HABANALABS_PATH)/board/habanalabs/common/post-image-fit-arc.sh $(BR2_EXTERNAL_HABANALABS_PATH)/board/habanalabs/gaudi/post-image-spi-arc_global.sh $(BR2_EXTERNAL_HABANALABS_PATH)/board/habanalabs/common/post-image-agents.sh $(BR2_EXTERNAL_HABANALABS_PATH)/board/habanalabs/$(HL_DEV)/create-fake-fuse-image.sh"
 # BR2_PACKAGE_BUSYBOX is not set
 # BR2_PACKAGE_IFUPDOWN_SCRIPTS is not set
 # BR2_TARGET_ROOTFS_TAR is not set
diff --git a/configs/gaudi3_arc_defconfig b/configs/gaudi3_arc_defconfig
index 6e5b96e..43376ba 100644
--- a/configs/gaudi3_arc_defconfig
+++ b/configs/gaudi3_arc_defconfig
@@ -12,7 +12,7 @@
 BR2_INIT_NONE=y
 BR2_SYSTEM_BIN_SH_NONE=y
 # BR2_TARGET_GENERIC_GETTY is not set
-BR2_ROOTFS_POST_IMAGE_SCRIPT="$(BR2_EXTERNAL_HABANALABS_PATH)/board/habanalabs/common/post-image-fit-arc.sh $(BR2_EXTERNAL_HABANALABS_PATH)/board/habanalabs/$(HL_DEV)/gaudi3-post-image-spi-arc_global.sh $(BR2_EXTERNAL_HABANALABS_PATH)/board/habanalabs/common/post-image-agents.sh $(BR2_EXTERNAL_HABANALABS_PATH)/board/habanalabs/$(HL_DEV)/create-fake-fuse-image.sh"
+BR2_ROOTFS_POST_IMAGE_SCRIPT="$(BR2_EXTERNAL_HABANALABS_PATH)/board/habanalabs/common/post-image-fit-arc.sh $(BR2_EXTERNAL_HABANALABS_PATH)/board/habanalabs/gaudi/post-image-spi-arc_global.sh $(BR2_EXTERNAL_HABANALABS_PATH)/board/habanalabs/common/post-image-agents.sh $(BR2_EXTERNAL_HABANALABS_PATH)/board/habanalabs/$(HL_DEV)/create-fake-fuse-image.sh"
 # BR2_PACKAGE_BUSYBOX is not set
 # BR2_PACKAGE_IFUPDOWN_SCRIPTS is not set
 # BR2_TARGET_ROOTFS_TAR is not set
diff --git a/configs/gaudi3_arc_onix_defconfig b/configs/gaudi3_arc_onix_defconfig
index aa78bbd..e37d127 100644
--- a/configs/gaudi3_arc_onix_defconfig
+++ b/configs/gaudi3_arc_onix_defconfig
@@ -12,7 +12,7 @@
 BR2_INIT_NONE=y
 BR2_SYSTEM_BIN_SH_NONE=y
 # BR2_TARGET_GENERIC_GETTY is not set
-BR2_ROOTFS_POST_IMAGE_SCRIPT="$(BR2_EXTERNAL_HABANALABS_PATH)/board/habanalabs/common/post-image-fit-arc.sh $(BR2_EXTERNAL_HABANALABS_PATH)/board/habanalabs/$(HL_DEV)/gaudi3-post-image-spi-arc_global.sh $(BR2_EXTERNAL_HABANALABS_PATH)/board/habanalabs/common/post-image-agents.sh $(BR2_EXTERNAL_HABANALABS_PATH)/board/habanalabs/$(HL_DEV)/create-fake-fuse-image.sh"
+BR2_ROOTFS_POST_IMAGE_SCRIPT="$(BR2_EXTERNAL_HABANALABS_PATH)/board/habanalabs/common/post-image-fit-arc.sh $(BR2_EXTERNAL_HABANALABS_PATH)/board/habanalabs/gaudi/post-image-spi-arc_global.sh $(BR2_EXTERNAL_HABANALABS_PATH)/board/habanalabs/common/post-image-agents.sh $(BR2_EXTERNAL_HABANALABS_PATH)/board/habanalabs/$(HL_DEV)/create-fake-fuse-image.sh"
 # BR2_PACKAGE_BUSYBOX is not set
 # BR2_PACKAGE_IFUPDOWN_SCRIPTS is not set
 # BR2_TARGET_ROOTFS_TAR is not set
diff --git a/configs/gaudi3_arc_palad_defconfig b/configs/gaudi3_arc_palad_defconfig
index aa78bbd..e37d127 100644
--- a/configs/gaudi3_arc_palad_defconfig
+++ b/configs/gaudi3_arc_palad_defconfig
@@ -12,7 +12,7 @@
 BR2_INIT_NONE=y
 BR2_SYSTEM_BIN_SH_NONE=y
 # BR2_TARGET_GENERIC_GETTY is not set
-BR2_ROOTFS_POST_IMAGE_SCRIPT="$(BR2_EXTERNAL_HABANALABS_PATH)/board/habanalabs/common/post-image-fit-arc.sh $(BR2_EXTERNAL_HABANALABS_PATH)/board/habanalabs/$(HL_DEV)/gaudi3-post-image-spi-arc_global.sh $(BR2_EXTERNAL_HABANALABS_PATH)/board/habanalabs/common/post-image-agents.sh $(BR2_EXTERNAL_HABANALABS_PATH)/board/habanalabs/$(HL_DEV)/create-fake-fuse-image.sh"
+BR2_ROOTFS_POST_IMAGE_SCRIPT="$(BR2_EXTERNAL_HABANALABS_PATH)/board/habanalabs/common/post-image-fit-arc.sh $(BR2_EXTERNAL_HABANALABS_PATH)/board/habanalabs/gaudi/post-image-spi-arc_global.sh $(BR2_EXTERNAL_HABANALABS_PATH)/board/habanalabs/common/post-image-agents.sh $(BR2_EXTERNAL_HABANALABS_PATH)/board/habanalabs/$(HL_DEV)/create-fake-fuse-image.sh"
 # BR2_PACKAGE_BUSYBOX is not set
 # BR2_PACKAGE_IFUPDOWN_SCRIPTS is not set
 # BR2_TARGET_ROOTFS_TAR is not set
